core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java

```diff
@@ -211,7 +220,13 @@ protected long writeMaxFileSize() {
   }
 
   private long sizeOfInputFiles(List<FileScanTask> group) {
-    return group.stream().mapToLong(FileScanTask::length).sum();
+    return group.stream().mapToLong(this::sizeOfInputFile).sum();
+  }
+
+  private long sizeOfInputFile(FileScanTask file) {
```

Comment:
@RussellSpitzer, what do you think about this change to `BinPackStrategy`?

I'm not sure that we want to make this change to the file rewrite. While delete files are relevant to how this will run, the bin packing here is not to balance work in this job, it is to pack data into files so that work is balanced in future jobs.

I think it is probably a mistake to change this class.

Reply:
I think the `file groups` here depends the rewrite job scale, it should take into account how much delete files cost in each rewrite job, otherwise there could be 'job-level' unbalanced issue.
@rdblue @RussellSpitzer what do you think?

Reply:
I don't think this should be based on deletes.

Reply:
@rdblue OK, I revert the code change of `BinPackStrategy`, just introduce deletes based `weightFunc` in `TableScanUtil`, please have a further review, thank you